float32 x
float32 y
float32 res_x
float32 res_y
float32 random_angle_left
float32 random_angle_right
